### DISCUSSION TITLE
I need Ctrl + C to connect to my wsl

### DISCUSSION THREAD

**Discussion by TheMakarov** [microsoft/WSL#6611] [General] [answered]
# Environment
![image](https://user-images.githubusercontent.com/58312936/109626679-1deb9080-7b41-11eb-82e4-63fedc94c8e0.png)
WSL 2 Kali Linux
![image](https://user-images.githubusercontent.com/58312936/109626776-3b205f00-7b41-11eb-972c-d880f4cf7bec.png)

# Steps to reproduce
I open WSL no matter from the icon or from the power shell and then i need to wait a little and hit Ctrl + C to login to my account 
and this doesn't give me any issues when i login as root 

# Standard user login
Picture 1
![image](https://user-images.githubusercontent.com/58312936/109626972-6e62ee00-7b41-11eb-99e7-1477632ebd5b.png)
Picture 2
![image](https://user-images.githubusercontent.com/58312936/109627237-b08c2f80-7b41-11eb-9ca5-c3778f771387.png)

#root login 
Picture 3
![image](https://user-images.githubusercontent.com/58312936/109627057-833f8180-7b41-11eb-89b2-ad6b21be418e.png)

#without hitting Ctrl + C
Picture 4 in powershell  
![image](https://user-images.githubusercontent.com/58312936/109627500-fa751580-7b41-11eb-8199-6ccb5c1f9e7e.png)
it wait for a while then 
Picture 5
![image](https://user-images.githubusercontent.com/58312936/109627543-0365e700-7b42-11eb-9955-6e7da414e839.png)
 
Picture 6
For the regular terminal it leaves after a while
![image](https://user-images.githubusercontent.com/58312936/109627645-1e385b80-7b42-11eb-96c3-9d73a8fdff02.png)

# Last installs

i recently installs kex or win-kex for viewing gui can this effect 
and thanks after all

**Comment by therealkenc** [comment, marked as answer]
Something is blocking in your `fish` shell's starup files. The interrupt unblocks whatever it is waiting on. Starting WSL with `wsl -d kali -u root` does not block because it (almost certainly) has `/bin/bash` as the shell and unadulterated startup files.

**Reply by TheMakarov** [reply]
Yes surely it was a mistake that i added a stupid code to config.fish file 
Thanks my friend a lot
